foundation/device/device-client/src/main/kotlin/com/pingidentity/device/client/DeviceClient.kt

```diff
@@ -0,0 +1,299 @@
+        device: Device,
+    ): String {
+        val uri = composeBaseUrl(config)
+            .appendEncodedPath(device.urlSuffix)
+            .appendEncodedPath(device.id)
+        return uri.build().toString()
+    }
+
+    /**
+     * Fetch a list of devices of type [T] from the server.
+     *
+     * @param config The client configuration.
+     * @param path The API path for the device type.
+     * @return List of devices of type [T].
+     */
+    private suspend inline fun <reified T : Device> getDeviceList(
+        config: DeviceClientConfig,
+        path: String,
+    ): List<T> {
+        val request = httpClient.prepareRequest {
+            url(composeUrlForDeviceList(config, path))
+            header(config.cookieName, config.ssoTokenString ?: "")
+            header("Content-Type", "application/json")
+            header("Accept-API-Version", "resource=1.0")
+            header("x-requested-platform", "Android")
+            method = Get
+        }
+        val body = request.execute().bodyAsText()
+        val jsonObject = Json.parseToJsonElement(body).jsonObject
+        val result = jsonObject["result"]?.jsonArray
+        return result?.map {
+            val obj = it.jsonObject
+            Json.decodeFromString(obj.toString())
+        } ?: emptyList()
+    }
+
+    /**
+     * Delete a device of type [T] on the server.
+     *
+     * @param config The client configuration.
+     * @param device The device to delete.
+     * @return The HTTP response from the server.
+     */
+    private suspend inline fun <reified T : Device> deleteDevice(
+        config: DeviceClientConfig,
+        device: T,
+    ): HttpResponse {
+        val urlString = composeUrlForDevice(
+            config = config,
+            device = device,
+        )
+        val request = httpClient.prepareRequest {
+            url(urlString)
+            header(config.cookieName, config.ssoTokenString ?: "")
+            header("Content-Type", "application/json")
+            header("Accept-API-Version", "resource=1.0")
+            header("x-requested-platform", "Android")
+            method = Delete
+        }
+        return request.execute()
+    }
+
+    /**
+     * Update a device of type [T] on the server.
+     *
+     * @param config The client configuration.
+     * @param device The device to update.
+     * @return The HTTP response from the server.
+     */
+    private suspend inline fun <reified T : Device> updateDevice(
+        config: DeviceClientConfig,
+        device: T,
+    ): HttpResponse {
+        val request = httpClient.prepareRequest {
+            url(composeUrlForDevice(
+                config = config,
+                device = device,
+            ))
+            header(config.cookieName, config.ssoTokenString ?: "")
+            header("Content-Type", "application/json")
+            header("Accept-API-Version", "resource=1.0")
+            header("x-requested-platform", "Android")
```

Comment:
The Network module will inject this, does not need to include x-requested-platform